File: 2744-Find-Maximum-Number-of-String-Pairs/o_n.py
# https://leetcode.com/problems/find-maximum-number-of-string-pairs/submissions/1029027176/

class Solution:
    def maximumNumberOfStringPairs(self, words: List[str]) -> int:
        s = set()
        count = 0
        for word in words:
            if word in s:
                count += 1
            s.add(word[::-1])
        return count

File: 2744-Find-Maximum-Number-of-String-Pairs/o_n.js
// https://leetcode.com/problems/find-maximum-number-of-string-pairs/submissions/1029029425/
/**
 * @param {string[]} words
 * @return {number}
 */
var maximumNumberOfStringPairs = function(words) {
    const s = new Set();
    let count = 0;
    for(const word of words){
        if(s.has(word)){
            count += 1;
        }
        s.add(word.split("").reverse().join(""));
    }
    return count;
};
